packages/credentials/src/ctype/CTypeLoader.ts

```diff
@@ -32,9 +37,69 @@ export function newCachingCTypeLoader(
   })
 
   async function getCType(id: ICType['$id']): Promise<ICType> {
```

Comment:
Randomly adding more soundness checks?

Reply:
I moved them from the code for loading nested ctype definitions, as they seemed misplaced there.

Reply:
Ok fair enough.